Added support for uploading multiple files, implemented analysis for multiple files, returned confidence levels for answers, and added feature to hide low confidence answers.

DanielTea opened the pull request:
- Added a patch that allows users to upload multiple files at once, which improves the user experience by saving time and 
effort.

-Implemented the ability to analyze multiple files simultaneously, providing more flexibility and convenience to the user.

-The system now returns a confidence level for the answer provided, giving users a sense of the reliability of the information.
Implemented a feature to hide answers with low confidence levels, which ensures the accuracy and credibility of the results.

mmz-001 commented:
Thanks for the PR. I'll go through this and check if everything is fine before we can merge. (Might take some time though)

DanielTea commented:
> Thanks for the PR. I'll go through this and check if everything is fine before we can merge. (Might take some time though)

Hey thanks! I‘m not sure if we rather should merge it to dev first, cause I‘m a little inconfident if the prompt re engineering correct.

mmz-001 commented:
> Hey thanks! I‘m not sure if we rather should merge it to dev first, cause I‘m a little inconfident if the prompt re engineering correct.

After we've got everything working in this PR we can merge it to main. 

What I suggest is that we split this PR into two, one for handling multiple files and one for the confidence levels so we can work on them separately. We could first work on the confidence level feature and move on to handling multiple files (which is the harder one to get right).

Can you split your changes into two branches and create two PRs?

DanielTea commented:
[Add multiple file upload, contributors, and parallelize embedding](https://github.com/mmz-001/knowledge_gpt/pull/33/commits/05475edec440c682a4c875b861b750784a4c94c2) 
[05475ed](https://github.com/mmz-001/knowledge_gpt/pull/33/commits/05475edec440c682a4c875b861b750784a4c94c2)
- Added multiple file upload feature using Streamlit file uploader
- Added loading bar using tqdm library to give feedback to the user on indexing progress
- Added section to sidebar displaying contributors to the project, retrieved using the GitHub API
- Parallelized document embedding function using concurrent.futures module
- Added progress bar using tqdm module to provide visual feedback on embedding progress
- reverted changes in the prompting
